presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -14,65 +19,83 @@ class AnonymizerEngine:
     and replaces the PII entities with the desired anonymizers.
     """
 
-    logger = logging.getLogger("presidio-anonymizer")
-    builtin_anonymizers = {
-        cls.anonymizer_name(cls): cls for cls in Anonymizer.__subclasses__()
-    }
-
-    def __init__(
-            self,
-    ):
-        """Handle text replacement for PIIs with requested anonymizers.
-
-        :param data: a map which contains the anonymizers, analyzer_results and text
-        """
+    def __init__(self):
+        self.logger = logging.getLogger("presidio-anonymizer")
 
-    def anonymize(self, engine_request: AnonymizerRequest) -> str:
+    def anonymize(self, text: str, analyzer_results: List[AnalyzerResult],
```

Comment:
If we support anonymizers which don't require params (like replace or redact), maybe we should have the `anonymizers_config` as optional. It would make the out-of-the-box experience much simpler. For example:

```python
from presidio_anonymizer import AnonymizerEngine
from presidio_anonymizer.entities import AnalyzerResult

text = "Dan is my name"
result = AnalyzerResult(entity_type="NAME", start=0, end=3, score=0.5)

engine = AnonymizerEngine()
engine.anonymize(text, [result])
```

Reply:
Done.